lib/httpClient.ts

```diff
@@ -22,6 +23,14 @@ export default abstract class HTTPClient {
             this.client.defaults.headers.common['X-Axiom-Org-Id'] = orgID;
         }
 
+        // We should only retry in the case the status code is >= 500, anything below isn't worth retrying.
+        axiosRetry(this.client, {
+            retryDelay: axiosRetry.exponentialDelay,
+            retryCondition: (error: any) => {
+                return error.response.status >= 500;
```

Comment:
We also want to retry if there's a (retryable) network error, could you add that condition as well?

Reply:
other than 502,504? 
The only one that comes to mind is 408 (Request Timeout), is that what you mean?

Reply:
If the client can’t establish a connection (ie no internet), you won’t even get a response code

Reply:
thinking if something like this could work:

```js
axiosRetry(this.client, {
            retryDelay: axiosRetry.exponentialDelay,
            retryCondition: (error: any) => {
                return isNetworkOrIdempotentRequestError(error) || error.response.status >= 500;
            },
        })
```

Reply:
I tried to test it with:

```js
    it('Retries requests on timeout', async function () {
        const scope = nock('http://axiom-node-retries.dev.local');
        scope.get('/api/v1/datasets').delayConnection(35000).reply(200, [{name: 'test'}]);

        const resp = await client.datasets.list();
        expect(scope.isDone()).eq(true);
        expect(resp.length).eq(1);
    }).timeout(50000);
```

but it fails. I also realize that timeout is too long for the test.

Reply:
switched to `isNetworkError(error) || isRetryableError(error) || (error.response && error.response.status >= 500);`